situp-core/src/main/java/com/amazon/situp/parser/PipelineParser.java

```diff
@@ -74,24 +74,37 @@ private void buildPipelineFromConfiguration(
         final PipelineConfiguration pipelineConfiguration = pipelineConfigurationMap.get(pipelineName);
         LOG.info("Building pipeline [{}] from provided configuration", pipelineName);
         try {
+            final int processorThreads = getWorkersOrDefault(pipelineConfiguration.getWorkers());
             final PluginSetting sourceSetting = pipelineConfiguration.getSourcePluginSetting();
+            updatePluginSettings(sourceSetting, pipelineName, processorThreads);
             final Optional<Source> pipelineSource = getSourceIfPipelineType(pipelineName, sourceSetting,
                     pipelineMap, pipelineConfigurationMap);
             final Source source = pipelineSource.orElseGet(() -> SourceFactory.newSource(sourceSetting));
 
             LOG.info("Building buffer for the pipeline [{}]", pipelineName);
-            final Buffer buffer = getBufferOrDefault(pipelineConfiguration.getBufferPluginSetting());
+            final PluginSetting bufferSetting = pipelineConfiguration.getBufferPluginSetting();
+            Buffer buffer;
+            if (bufferSetting != null) {
+                updatePluginSettings(bufferSetting, pipelineName, processorThreads);
+                buffer = BufferFactory.newBuffer(bufferSetting);
+            } else {
+                buffer = new BlockingBuffer(pipelineName);
+            }
 
             LOG.info("Building processors for the pipeline [{}]", pipelineName);
             final List<Processor> processors = pipelineConfiguration.getProcessorPluginSettings().stream()
-                    .map(ProcessorFactory::newProcessor)
+                    .map(processorSetting -> {
+                        updatePluginSettings(processorSetting, pipelineName, processorThreads);
+                        return ProcessorFactory.newProcessor(processorSetting); })
                     .collect(Collectors.toList());
-            final int processorThreads = getWorkersOrDefault(pipelineConfiguration.getWorkers());
             final int readBatchDelay = getDelayOrDefault(pipelineConfiguration.getReadBatchDelay());
 
             LOG.info("Building sinks for the pipeline [{}]", pipelineName);
             final List<Sink> sinks = pipelineConfiguration.getSinkPluginSettings().stream()
-                    .map(this::buildSinkOrConnector).collect(Collectors.toList());
+                    .map(sinkSetting -> {
+                        updatePluginSettings(sinkSetting, pipelineName, processorThreads);
+                        return buildSinkOrConnector(sinkSetting); })
+                    .collect(Collectors.toList());
```

Comment:
Instead of doing these updates separately, would doing something like below make it better structured.

In the `PipelineConfiguration.java` add the below method

```
pipelineconfiguration.updateCommonPipelineConfig(pipelinName) {
  updatePluginSettings(this.buffer, pipelineName)
 sinkSettings.foreach(i -> updatePluginSetting(i, pipelineName));
processorSettings.foreach(i -> updatePluginSetting(i, pipelineName));
}
```

and invoke this method in the `pipelineparser.java`

Reply:
+1

Reply:
discussed offline about the need and approach